src/stan/variational/advi.hpp

```diff
@@ -374,13 +553,29 @@ namespace stan {
       BaseRNG& rng_;
       int n_monte_carlo_grad_;
       int n_monte_carlo_elbo_;
-      double eta_adagrad_;
       int eval_elbo_;
       int n_posterior_samples_;
       std::ostream* print_stream_;
       std::ostream* out_stream_;
       std::ostream* diag_stream_;
 
+      double circ_buff_median_(const boost::circular_buffer<double>& cb) const {
```

Comment:
It looks like this function is duplicated. Can you:
1. consolidate the cut and paste.
2. test it as a unit test? what happens when cb is length 0?